File: app/src/main/java/com/example/bookapp/Adapters/SuggestionsAdapter.java
package com.example.bookapp.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.bookapp.databinding.LayoutPostItemBinding;
import com.example.bookapp.fragments.ExpandedItemFragmentDirections;
import com.example.bookapp.models.Post;

import java.util.ArrayList;

public class SuggestionsAdapter extends RecyclerView.Adapter<SuggestionsAdapter.ViewHolder> {
    private ArrayList<Post> data;
    private Context context;

    public SuggestionsAdapter(ArrayList<Post> data) {
        this.data = data;
    }

    public ArrayList<Post> getData() {
        return data;
    }
    public void setData(ArrayList<Post> data){
        this.data = data;
        notifyDataSetChanged();

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        LayoutPostItemBinding layoutSuggestionItemBinding = LayoutPostItemBinding
                .inflate(layoutInflater, parent, false);
        return new ViewHolder(layoutSuggestionItemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Post currentPost = data.get(position);
        holder.bindData(currentPost);
        Glide.with(context)
                .load(currentPost.getPostImage())
                .into(holder.binding.suggestionImage);
        holder.binding.getRoot().setOnClickListener(view -> {
            NavDirections action = ExpandedItemFragmentDirections.actionGlobalExpandedItemFragment(currentPost.getPostID());
            Navigation.findNavController(holder.binding.getRoot()).navigate(action);
        });
    }

    @Override
    public int getItemCount() {
        return data.size();
    }


    class ViewHolder extends RecyclerView.ViewHolder {
        LayoutPostItemBinding binding;

        ViewHolder(@NonNull LayoutPostItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bindData(Post post) {
            binding.setSuggestionPost(post);
        }
    }
}

File: app/src/main/java/com/example/bookapp/Adapters/SuggestionsAdapter.kt
package com.example.bookapp.Adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.navigation.NavDirections
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.bookapp.databinding.LayoutPostItemBinding
import com.example.bookapp.fragments.ExpandedPostFragmentDirections
import com.example.bookapp.models.LowDataPost
import java.util.*

class SuggestionsAdapter : RecyclerView.Adapter<SuggestionsAdapter.ViewHolder>() {
    private var context: Context? = null
    var data: ArrayList<LowDataPost> = ArrayList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        val layoutInflater = LayoutInflater.from(parent.context)
        val layoutSuggestionItemBinding = LayoutPostItemBinding
                .inflate(layoutInflater, parent, false)
        return ViewHolder(layoutSuggestionItemBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentPost = data[position]
        holder.bindData(currentPost)
        Glide.with(context!!)
                .load(currentPost.image)
                .into(holder.binding.suggestionImage)
        holder.binding.root.setOnClickListener {
            val action: NavDirections = ExpandedPostFragmentDirections.actionGlobalExpandedItemFragment(currentPost.id)
            Navigation.findNavController(holder.binding.root).navigate(action)
        }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    inner class ViewHolder(var binding: LayoutPostItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bindData(post: LowDataPost) {
            binding.suggestionPost = post
        }

    }

}